src/main/java/uk/gov/hmcts/cmc/claimstore/services/SettlementAgreementService.java

```diff
@@ -70,6 +71,10 @@ public Claim countersign(Claim claim, String authorisation) {
 
         eventProducer.createSettlementAgreementCountersignedEvent(updated);
         appInsights.trackEvent(SETTLEMENT_AGREEMENT_REACHED, REFERENCE_NUMBER, updated.getReferenceNumber());
+        if (settlement.isSettlementThroughAdmissions()) {
+            appInsights.trackEvent(SETTLEMENT_AGREEMENT_REACHED_BY_ADMISSION, REFERENCE_NUMBER,
+                updated.getReferenceNumber());
```

Comment:
This code will also fire two events in case of settlement by admissions. Once outside if block and second one inside if block. Please correct this.

Reply:
```
AppInsightsEvent appInsightsEvent = settlement.isSettlementThroughAdmissions() ? SETTLEMENT_AGREEMENT_REACHED_BY_ADMISSION : SETTLEMENT_AGREEMENT_REACHED;
 appInsights.trackEvent(appInsightsEvent, REFERENCE_NUMBER,
                updated.getReferenceNumber());
```